runtime/src/iree/base/internal/dynamic_library_win32.c

```diff
@@ -319,6 +319,33 @@ iree_status_t iree_dynamic_library_lookup_symbol(
   return iree_ok_status();
 }
 
+iree_status_t iree_dynamic_library_get_symbol_path(
+    void* symbol, iree_string_builder_t* out_path) {
+  HMODULE hm = NULL;
+  if (GetModuleHandleEx(GET_MODULE_HANDLE_EX_FLAG_FROM_ADDRESS |
+                            GET_MODULE_HANDLE_EX_FLAG_UNCHANGED_REFCOUNT,
+                        (LPCSTR)symbol, &hm) == 0) {
+    return iree_make_status(IREE_STATUS_NOT_FOUND);
+  }
+
+  iree_string_builder_reset(out_path);
+  IREE_RETURN_IF_ERROR(iree_string_builder_reserve(out_path, 64));
+  while (1) {
+    out_path->size =
```

Comment:
don't overwrite the builder, but just append it - that makes it possible to compose things and not need unique builders

Reply:
Ok, added helpers to string builder (`iree_string_builder_reserve_for_append` and `iree_string_builder_commit_append`) and used them here. Since this is such a common WIN32 idiom, I didn't want to just open code it. Have dropped knives on my toes with that in the past.